libfwupd/fwupd-device.c

```diff
@@ -254,6 +255,70 @@ fwupd_device_get_guid_default (FwupdDevice *device)
 }
 
 /**
+ * fwupd_device_get_icon_names:
+ * @device: A #FwupdDevice
+ *
+ * Gets the icon names to use for the device.
+ *
+ * Returns: (element-type utf8) (transfer none): an array of icon names
+ *
+ * Since: 0.9.8
+ **/
+GPtrArray *
+fwupd_device_get_icon_names (FwupdDevice *device)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE (device);
+	g_return_val_if_fail (FWUPD_IS_DEVICE (device), NULL);
+	return priv->icon_names;
+}
+
+/**
+ * fwupd_device_has_icon_name:
+ * @device: A #FwupdDevice
+ * @icon_name: the name, e.g. "input-mouse"
+ *
+ * Finds out if the device has this specific icon name already.
+ *
+ * Returns: %TRUE if the icon name is found
+ *
+ * Since: 0.9.8
+ **/
+gboolean
+fwupd_device_has_icon_name (FwupdDevice *device, const gchar *icon_name)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE (device);
+	guint i;
+
+	g_return_val_if_fail (FWUPD_IS_DEVICE (device), FALSE);
+
+	for (i = 0; i < priv->icon_names->len; i++) {
+		const gchar *icon_name_tmp = g_ptr_array_index (priv->icon_names, i);
+		if (g_strcmp0 (icon_name, icon_name_tmp) == 0)
+			return TRUE;
+	}
+	return FALSE;
+}
```

Comment:
Is there a good reason to have `fwupd_device_has_icon_name` available outside of libfwupd?  It just seems to me it's better to keep this code as part of the function `fwupd_device_add_icon_name ` and then not expose it elsewhere.

Reply:
Agree.